File: code-interview/01.6/solution.js
function rotateImage(image) {
	for (var layer = 0; layer = image.length  / 2; layer++)
	{
		var first = layer;
		var last = image.length - layer - 1;
		for (var i = first; i < last; i ++)
		{
			var offset = i - first;
			var top = image[first][i];

			image[first][i] = image[last - offset][first];
			image[last-offset][first] = matrix[last][last - offset];
			image[last][last - offset] = matrix[i][last];
			matrix[i][last] = top;
		}
	}
	return image;
}

function printImage(image) {
	image.forEach(function(row, i) {
		console.log(row);
	})
}

tests = [[[1,2],[3,4]],
	[[1,2,3], [4,5,6],[7,8,9]]];

tests.forEach(function(test){
	console.log('-image-');
	printImage(test);
	test = rotateImage(test);
	console.log('-rotated-');
	printImage(test);

});

File: code-interview/01.6/solution.py
from __future__ import print_function

def rotateImage(image, direction):
	image = transposeImage(image)
	return image

def transposeImage(image):
	newImage = ()
	for i, row in enumerate(image):
		for j, col in enumerate(row):
			if (newImage == None):
					'''newImage[j] = ()'''
	return image

def printImage(image):
	for row in image:
		for col in row:
			print(col, end="")
		print("\n")

tests = (((1,2),(3,4)),
	((1,2,3), (4,5,6),(7,8,9)))

for test in tests:
	print('-image-')
	printImage(test)
	test = rotateImage(test, 'right')
	print('-rotated-')
	printImage(test)